src/cljs/main/broadfcui/page/workspace/summary/catalog/questions.cljs

```diff
@@ -109,49 +110,61 @@
+  (let [[related-id-prop related-label-prop] (library-utils/get-related-id+label-props library-schema property)
+        [related-id related-label clear-fields] (handle-differences (library-utils/get-related-id+label (:attributes @state) library-schema property))
+        clear (fn []
+                (apply swap! state update :attributes dissoc property [related-id-prop related-label-prop])
+                ((@refs (name property)) :set-value ""))]
     [:div {:style {:marginBottom "0.75em"}}
-     [Autosuggest
-      {:value value-nullsafe
-       :inputProps {:placeholder (:inputHint prop)
-                    :data-test-id property
-                    :disabled disabled}
-       :url "/autocomplete/"
-       :service-prefix "/duos"
-       :get-value #(.-label %)
-       :renderSuggestion (fn [suggestion]
-                           (react/create-element
-                            [:div {}
-                             [:div {:style {:lineHeight "1.5em"}}
-                              (.-label suggestion)
-                              [:small {:style {:float "right"}} (.-id suggestion)]]
-                             [:small {:style {:fontStyle "italic"}}
-                              (.-definition suggestion)]
-                             [:div {:style {:clear "both"}}]]))
-       :highlightFirstSuggestion false
-       :onSuggestionSelected (fn [_ suggestion]
-                               (let [suggestion (js->clj (.-suggestion suggestion) :keywordize-keys true)
-                                     {:keys [id label]} suggestion
-                                     [related-id-prop related-label-prop] (map (comp keyword #(% prop)) [:relatedID :relatedLabel])]
-                                 (swap! state update :attributes assoc
-                                        related-id-prop id
-                                        related-label-prop label)))
-       :on-change set-property
-       :on-submit #(when (empty? %) (clear))
-       :theme {:input (colorize {:width "100%" :marginBottom 0})
-               :suggestionsContainerOpen {:marginTop -1 :width "100%"}}}]
+     (if (= (:type prop) "array")
+       [:div {}
+        (ontology/render-multiple-ontology-selections
+         {:onClick (fn [id label]
+                     (swap! state update-in [:attributes related-id-prop] library-utils/remove-from-comma-separated-strings id)
+                     (swap! state update-in [:attributes related-label-prop] library-utils/remove-from-comma-separated-strings label))
+          :selection-map (library-utils/zip-comma-separated-strings related-id related-label)})
+        (ontology/create-ontology-autosuggest
+         {:on-suggestion-selected
+          (fn [{:keys [id label]}]
+            (if clear-fields
+              (do
+                (swap! state update :attributes assoc related-id-prop id)
+                (swap! state update :attributes assoc related-label-prop label))
```

Comment:
There are 3 double swap!s here. This one can be trivially combined, and I'd prefer to see the others combined too.